### PR TITLE
Add support for admins adding and removing whitelists

### PR DESCRIPTION
**Pull request by nvonpentz**
* Add publisher_whitelist_updates table

* Add PublisherWhitelistUpdates model, controller

* Expose UI to enable admins to use API

* Add test for PublisherWhitelistUpdatesController

## Pull Request Name
View status
image

Update status
image

#### Features

Enter a brief summary of the Pull Request - Features, motivations, and issues it #closes
#### How To Test

1. Step one
2. Step two
3. Step three

#### Pull Request Checklist

- [ ] Adequate test coverage exists to prevent regressions -- [Guide to Testing](https://guides.rubyonrails.org/testing.html)
- [ ] XSS is mitigated -- [Guide to XSS Prevention](https://guides.rubyonrails.org/security.html#cross-site-scripting-xss)
- [ ] No raw SQL -- [Guide to SQL Injection Prevention](https://guides.rubyonrails.org/security.html#sql-injection)
- [ ] UI/UX is responsive -- [Guide to Responsive UI/UX](https://developers.google.com/web/fundamentals/design-and-ux/responsive/)
- [ ] Integrated Matomo for new UI elements -- [Guide to Matomo](https://developer.matomo.org/guides/integrate-introduction)
- [ ] Passes checklist for Progressive Web App -- [Guide to PWAs](https://developers.google.com/web/progressive-web-apps/checklist)

### CODE REVIEW

File: app/controllers/admin/publishers/publisher_whitelist_updates_controller.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+class Admin::Publishers::PublisherWhitelistUpdatesController < Admin::PublishersController
+  def create
+    note = @publisher.notes.create!(note: params[:note], created_by_id: current_publisher.id)
+    PublisherWhitelistUpdate.create!(publisher: @publisher, enabled: params[:enable], publisher_note: note)
```
Comment (1nezer0) [inline app/controllers/admin/publishers/publisher_whitelist_updates_controller.rb:4]:
```suggestion
    PublisherWhitelistUpdate.create!(publisher: @publisher, enabled: params[:enable].present?, publisher_note: note)
```

File: db/migrate/20201215124546_add_publisher_whitelist_updates.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+class AddPublisherWhitelistUpdates < ActiveRecord::Migration[6.0]
+  def change
+    create_table :publisher_whitelist_updates, id: :uuid, default: -> { "uuid_generate_v4()"}, force: :cascade do |t|
+      t.belongs_to :publisher, type: :uuid, index: true, unique: true, null: false
+      t.references :publisher_note, type: :uuid
+      t.boolean :enabled, null: false
+      t.timestamps
```
Comment (1nezer0) [inline db/migrate/20201215124546_add_publisher_whitelist_updates.rb:7]: Would you add indexes to the timestamps?

File: app/controllers/admin/publishers/publisher_whitelist_updates_controller.rb (hunk 1 above)
Comment (nvonpentz) [reply app/controllers/admin/publishers/publisher_whitelist_updates_controller.rb:4]: This param is probably poorly named.  If it's `false` then we disable the whitelist, if it's `true` then we enable the whitelist. There is a non null constraint in the database, so this should always be set.

File: db/migrate/20201215124546_add_publisher_whitelist_updates.rb (hunk 2 above)
Comment (nvonpentz) [reply db/migrate/20201215124546_add_publisher_whitelist_updates.rb]: Added

File: app/controllers/admin/publishers/publisher_status_updates_controller.rb [hunk 3]
```diff
@@ -6,6 +6,10 @@ def index
   end
 
   def create
+    if @publisher.last_whitelist_update&.enabled && [PublisherStatusUpdate::NO_GRANTS, PublisherStatusUpdate::SUSPENDED].include?(status)
```
Comment (1nezer0) [inline app/controllers/admin/publishers/publisher_status_updates_controller.rb:9]: `status` should be defined

File: app/controllers/admin/publishers/publisher_whitelist_updates_controller.rb (hunk 1 above)
Comment (1nezer0) [reply app/controllers/admin/publishers/publisher_whitelist_updates_controller.rb:4]: `params[:enabled].present?` just handles the edge case of `params[:enabled]` being `nil` to be `false`

Comment (nvonpentz) [reply app/controllers/admin/publishers/publisher_whitelist_updates_controller.rb]: I committed this and the test actually started failing.  Because what happens here is if we supply an explicit `false`, then `params[:enable].present?` will be `true` which isn't what we want

File: app/controllers/admin/publishers/publisher_status_updates_controller.rb (hunk 3 above)
Comment (nvonpentz) [reply app/controllers/admin/publishers/publisher_status_updates_controller.rb]: Used params[:publisher_status] instead

**Review by yachtcaptain23 — APPROVED** [review]
